Nybble/nybble_modules/create_alert.py

```diff
@@ -0,0 +1,154 @@
+from json import JSONEncoder
+from math import ceil
+from pydantic import BaseModel, Field
+import requests
+from requests import Response
+from nybble_modules import NybbleAction
+
+
+class CreateAlertArguments(BaseModel):
+    alert_data: dict = Field(..., description="Received alert, from Sekoia 'Get Alert' action")
+    events: list[dict] = Field(..., description="Related Events, from Sekoia 'Get The Alert Events' action")
+
+
+class CreateAlertResults(BaseModel):
+    status: bool
+    details: str
+
+
+class NybbleAlert(dict):
+    """
+    Final Object sent to Nybble Hub
+    """
+
+    alert_processing_time: str
+    alert_fields: dict[str, str]
+    alert_event_original: dict
+    alert_id: str
+    alert_level: int
+    rule_status: str
+    rule_name: str
+    rule_tags: list[str]
+    rule_references: list[str]
+    sigma_rule_id: str
+    mssp_client: str
+    custom_fields_visible: dict
+    custom_fields_hidden: dict
+
+
+class CreateAlertAction(NybbleAction):
+    """
+    Action to create an alert into Nybble Hub
+    """
+
+    name = "Create Alert"
+    description = "Create an Alert into Nybble Hub"
+    results_model = CreateAlertResults
+
+    def _generateTags(self, rule_tags) -> list[str]:
+        """
+        generate tag array from rule catalog
+        """
+        if len(rule_tags) == 0:
+            return []
+
+        nybble_tags = list[str]()
+        for tag in rule_tags:
+            nybble_tags.append(tag["name"])
+        return nybble_tags
+
+    def _generateFields(self, rule_event_fields: list[dict], event: dict) -> dict[str, str]:
+        """
+        generate fields from highlighted fields in rule + current event
+        """
+        nybble_fields = dict[str, str]()
+
+        #  Empty = host.name
+        if len(rule_event_fields) == 0:
+            return {"host.name": event["host.name"]}
+
+        for field in rule_event_fields:
+            cur_field = str(field["field"])
+            nybble_fields[cur_field] = event[cur_field]
+        return nybble_fields
+
+    def _cleanEventOriginal(self, original_event: dict) -> dict:
+        """
+        remove sekoiaio.xxx fields and rename timestamp -> @timestamp
+        """
+        cleaned_event = {k: v for k, v in original_event.items() if not str(k).startswith("sekoiaio")}
+        cleaned_event["@timestamp"] = cleaned_event.pop("timestamp")
+
+        return cleaned_event
+
+    def _getRuleDefinition(self, rule_uuid: str, api_key: str, base_url: str) -> dict:
+        """
+        Returns the rule from the rules catalog
+        """
+
+        url = f"{base_url}v1/sic/conf/rules-catalog/rules/{rule_uuid}"
+
+        response: Response = requests.get(url, headers={"Authorization": f"Bearer {api_key}"})
+        response.raise_for_status()
+        return response.json()
```

Comment:
It's the responsibility of the Sekoia.io automation module to allow users getting the definition of a rule in the playbook (We noticed that this action is missing from our module, and they plan to create soon).
I suggest removing this feature from the code and add an argument that will receive the content of the rule definition (like for the alert or the events).